src/Windows/Avalonia.Win32/CursorFactory.cs

```diff
@@ -87,5 +94,78 @@ public IPlatformHandle GetCursor(StandardCursorType cursorType)
 
             return rv;
         }
+
+        public IPlatformHandle CreateCursor(IBitmapImpl cursor, PixelPoint hotSpot)
+        {
+            using var source = LoadSystemDrawingBitmap(cursor);
+            using var mask = AlphaToMask(source);
+
+            var info = new UnmanagedMethods.ICONINFO
+            {
+                IsIcon = false,
+                xHotspot = hotSpot.X,
+                yHotspot = hotSpot.Y,
+                MaskBitmap = mask.GetHbitmap(),
+                ColorBitmap = source.GetHbitmap(),
+            };
+
+            return new PlatformHandle(
+                UnmanagedMethods.CreateIconIndirect(ref info),
+                PlatformConstants.CursorHandleType);
+        }
+
+        private SdBitmap LoadSystemDrawingBitmap(IBitmapImpl bitmap)
+        {
+            using var memoryStream = new MemoryStream();
+            bitmap.Save(memoryStream);
+            return new SdBitmap(memoryStream);
+        }
+
+        private unsafe SdBitmap AlphaToMask(SdBitmap source)
+        {
+            var dest = new SdBitmap(source.Width, source.Height, SdPixelFormat.Format1bppIndexed);
+
+            if (source.PixelFormat != SdPixelFormat.Format32bppArgb &&
```

Comment:
You allow both premultiplied and non-premultiplied alpha but don't do convert afterwards

Reply:
Lets's just allow non-premultipled alpha for now then.